kyo-core/js/src/main/scala/kyo/safePublish.scala

```diff
@@ -0,0 +1,12 @@
+package kyo
+
+import scala.annotation.StaticAnnotation
+
+/** Enforces Java Memory Model's safe publication guarantees for final fields (vals).
+ *
+ * When applied to a field or class, ensures strict JMM semantics rather than Scala Native's
+ * default relaxed memory model. Has no effect on JVM or JavaScript platforms.
+ *
+ * @see [[scala.scalanative.annotation.safePublish]]
+ */
+class safePublish extends StaticAnnotation
```

Comment:
I made the annotation available on all platforms so it can be used from a `shared` source folder.